draft-packages/tooltip/KaizenDraft/Tooltip/utils/isSemanticElement.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { ReactNode } from "react"
+
+// Noting that this is contingent on our displayName for our components -  this something we control
+const allowedDisplayNames = ["Button", "IconButton"]
+
+export function extractDisplayName(type: React.FunctionComponent): string {
+  return type.displayName || type.name || "Unknown"
+}
+
+/**
+ *  Validates implicit or explicitly semantic roles required to make `aria-describedby` announce predictably with screen readers
+ */
+export const isSemanticElement = (
```

Comment:
@dougmacknz also keen to get your eyes across this util :) holler if you want me to run you through this